core/src/main/scala/org/apache/spark/eventhubs/EventHubsConf.scala

```diff
@@ -561,6 +570,108 @@ final class EventHubsConf private (private val connectionStr: String)
             s"${PartitionPreferredLocationStrategy.values.mkString(",")}"))
   }
 
+  /**
+   * Set whether to use AAD auth to connect eventhubs.
+   * More info about this: https://docs.microsoft.com/en-us/azure/event-hubs/authorize-access-azure-active-directory
+   * There're 3 options for AadAuth: 1) Certificate; 2) Secret; 3) Callback. You'll also need to set
```

Comment:
Hi @duhuan , SJ and I discussed the options and we think we should just keep the Callback option here. In this way, the library is flexible to support various ways of obtaining an access token and users can provide their implementations to use either Certificate or Secret. 
We would like to have your AuthByCertCallBack and AuthBySecretCallBack implementations as samples that users can utilize in order to use the Callback option. It would be great if you could create a new "examples" directory under the root directory and move those implementations there as two examples of how the Callback option can be used for either Secret or Cert authentication.

Reply:
yeah, changed. please review the latest iteration.